### PR TITLE
Adjust citool to work with Travis build matrix.

### PR DESCRIPTION
**Pull request by rabbah**
The introduction of a Travis build matrix requires some changes to the `citool` to pull the right logs. This patch will allow for the build matrix index to be appended to the Travis job id to pull logs for a specific subjob.

```
> citool monitor 400722572.1
whisk.core.cli.test.WskRestEntitlementTests > Wsk Package Entitlement should not allow unauthorized subject to operate on private action FAILED
whisk.core.cli.test.WskRestEntitlementTests > Wsk Package Entitlement should reject update action in shared package not owned by authkey FAILED
whisk.core.cli.test.WskRestEntitlementTests > Wsk Package Action should invoke an action sequence from package FAILED
whisk.core.cli.test.WskRestEntitlementTests > Wsk Package Action should not allow invoke an action sequence with more than one component from package after entitlement change FAILED
whisk.core.cli.test.WskRestActionSequenceTests > Wsk Action Sequence should create, and get an action sequence FAILED
whisk.core.cli.test.WskRestBasicUsageTests > Wsk actions should reject action update for sequence with no components FAILED
whisk.core.cli.test.WskRestBasicUsageTests > Wsk actions should report error when creating an action with zip but without kind FAILED
whisk.core.database.test.CacheConcurrencyTests > the cache should support concurrent CRUD without bogus residual cache entries, iter 1 FAILED
system.basic.WskRestBasicTests > Wsk Action REST should create, update, get and list an action FAILED
system.basic.WskSequenceTests > Wsk Sequence should invoke a sequence with normal payload and payload with error field FAILED
system.basic.WskSequenceTests > Wsk Sequence should invoke a sequence with an enclosing sequence action FAILED
system.basic.WskSequenceTests > Wsk Sequence should replace atomic component in a sequence that is too long and report invoke error FAILED
system.basic.WskSequenceTests > Wsk Sequence should create and run a sequence in a package with parameters FAILED
system.basic.WskSequenceTests > Wsk Sequence should run a sequence with an action in a package binding with parameters FAILED
system.basic.WskSequenceTests > Wsk Sequence should stop execution of a sequence (with no payload) on error FAILED
system.basic.WskSequenceTests > Wsk Sequence should propagate execution error (timeout) from atomic action to sequence FAILED
system.basic.WskSequenceTests > Wsk Sequence should execute a sequence in blocking fashion and finish execution even if longer than blocking response timeout FAILED
system.basic.WskSequenceTests > Wsk Sequence should execute a sequence that is part of a rule and pass the trigger parameters to the sequence FAILED
system.basic.WskActionTests > Whisk actions should copy an action and invoke it successfully FAILED
system.basic.WskActionTests > Whisk actions should copy an action and ensure exec, parameters, and annotations copied FAILED
system.basic.WskActionTests > Whisk actions should add new parameters and annotations while copying an action FAILED
system.basic.WskActionTests > Whisk actions should support UTF-8 as input and output format FAILED
Done. Your build exited with 1.
```

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [x] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: tools/build/citool [hunk 1]
```diff
@@ -130,14 +135,25 @@ def getTravisHeaders():
...
+        except Exception:
+            print('Matrix id is not an integer as expected')
+
+        buildUrl = '%s/builds/%s' % (args.url, jobid)
         buildRes = request('get', buildUrl, headers = getTravisHeaders(), verbose = args.verbose)
         body = validateResponse(buildRes)
         try:
             body = json.loads(body)
-            job = body['build']['job_ids'][-1]
+            job = body['build']['job_ids'][matrix]
```
Comment (chetanmeh) [inline tools/build/citool]: It should be `matrix-1` as the matrix number start with 1 while job array is zero based.

File: tools/build/citool [hunk 2]
```diff
@@ -64,7 +68,8 @@ def parseArgs():
     parser = argparse.ArgumentParser(description='tool for analyzing logs from CI')
     subparsers = parser.add_subparsers(title='available commands', dest='cmd')
 
-    parser.add_argument('job', help='job number')
+    parser.add_argument('job', help='job number; for matrix jobs add the matrix index after a period')
```
Comment (chetanmeh) [inline tools/build/citool]: Possibly add an example `job number; for matrix jobs add the matrix index after a period e.g. 402061951.2`

**Comment by chetanmeh** [comment]
@rabbah As part of this change may be switch to [v3 API][1] as v2 [would be deprecated][2] in 2018

[1]: https://docs.travis-ci.com/user/developer/#API-V3
[2]: https://docs.travis-ci.com/api

File: tools/build/citool (hunk 1 above)
Comment (rabbah) [reply tools/build/citool]: k - I was doing the mapping on the cmd line but this make sense, will change.

**Comment by rabbah** [comment]
Thanks for the suggestions.

**Pull request merged**